lib/pipelineTemplateVersionFactory.js

```diff
@@ -0,0 +1,144 @@
+     * @param  {String}     config.description   Description of the template
+     * @param  {String}     config.maintainer    Maintainer's email
+     * @param  {Object}     config.config        Config of the screwdriver-template.yaml
+     * @param  {String}     config.pipelineId    pipelineId of the template
+     * @return {PipelineTemplateVersionModel}
+     */
+    createClass(config) {
+        return new PipelineTemplateVersionModel(config);
+    }
+
+    /**
+     * Create a new template of the correct version (See schema definition)
+     * @method create
+     * @param  {Object}     config               Config object
+     * @param templateMetaFactory
+     * @param  {Datastore}  config.datastore     Object that will perform operations on the datastore
+     * @param  {String}     config.name          The template name
+     * @param  {String}     config.namespace   The template namespace
+     * @param  {String}     config.version       Version of the template
+     * @param  {String}     config.description   Description of the template
+     * @param  {String}     config.maintainer    Maintainer's email
+     * @param  {Object}     config.config        Config of the screwdriver-template.yaml
+     * @param  {String}     config.pipelineId    pipelineId of the template
+     * @return {Promise}
+     */
+    async create(config, templateMetaFactory) {
+        const createTime = new Date().toISOString();
+        const [, configMajor, configMinor] = VERSION_REGEX.exec(config.version);
+
+        // get the template meta
+        let pipelineTemplateMeta = await templateMetaFactory.get({
+            name: config.name,
+            namespace: config.namespace
+        });
+
+        // if template meta doesn't exist, create one
+        if (!pipelineTemplateMeta) {
+            pipelineTemplateMeta = await templateMetaFactory.create({
+                pipelineId: config.pipelineId,
+                namespace: config.namespace,
+                name: config.name,
+                maintainer: config.maintainer,
+                createTime,
+                updateTime: createTime
+            });
+        }
+
+        let newVersion = configMinor ? `${configMajor}${configMinor}.0` : `${configMajor}.0.0`;
+
+        if (pipelineTemplateMeta.latestVersion) {
+            // list all the versions of the template
+            const pipelineTemplateVersions = await super.list({
+                params: {
+                    templateId: pipelineTemplateMeta.id,
+                    sort: 'descending',
+                    sortBy: 'createTime'
+                }
+            });
+
+            if (pipelineTemplateVersions.length > 0) {
+                // get latest version that have version starting with config.version
+                const pipelineTemplateVersion = pipelineTemplateVersions.find(template =>
+                    template.version.startsWith(`${configMajor}${configMinor || '.0'}`)
```

Comment:
`'2.34.5'.startsWith('2.3')` evaluates to `true`. So we may end up picking the wrong series.
```suggestion
                    template.version.startsWith(`${configMajor}${configMinor || '.0'}.`)
```